### ISSUE TITLE
Does the SC 1.4.3 Contrast minium exception apply to text outside a disabled control?

### ISSUE THREAD

**Issue by jha11y** [w3c/wcag#4874]
For the "Text or images of text that are part of an inactive user interface component" exception for WCAG 1.4.3 Contrast (Minimum), does the exception apply to the label for a control when it is outside of that control?

For example, a radio button that is disabled, does the text label does have to meet the contrast requirements of SC 1.4.3?

Would the exception apply to "helper text" (text underneath a text input field)? Error messages positioned like helper text?

I did check the other WCAG GitHub issues to see if my question had been answered, but I did not find this exact question.  I know text inside a disabled control (e.g., HTML button)  would be exempted.

**Comment by giacomo-petri** [comment]
This issue is currently stalled due to differing perspectives and a lack of agreement not only on the initial question, but also on the expected outcomes. The normative wording excludes inactive UIC, but the term "inactive" is not defined.

I've added the agenda label not necessary to solve this item but to understand how we want to move forward with it.

**Comment by patrickhlauke** [comment]
@kfranqueiro @mbgower it seems this issue was transferred in January to the `wcag3` repo, but it's still an issue that relates to WCAG 2.x. suggest it gets transferred back ... and if it was added to 3 purely because of "this is something we'll want to tackle in wcag3", then a new separate fresh issue is opened in `wcag3` about it. /cc @cwadamsoforacle

**Comment by mbgower** [comment]
@giacomo-petri are you able to provide some context why you moved this from To Do to For Discussion?

**Comment by bruce-usab** [comment]
Discussed briefly on backlog TF call 1/16 and we had concurrence with moving to our repo.  (As an FYI, this was [issue 160 in the WCAG3 repo](../../wcag3/issues/160).)

**Comment by aardrian** [comment]
To Scott's point, the only example in the Understanding doc is the button, whose label is part of the control. And because I have seen the impact of useful but illegible labels, I am largely in agreement with his take.

In an effort to unstick this, a suggestion of examples for the "inactive" exception:

* The text label of a button, the value of a disabled text field, and the options of a select menu are examples covered by this exception.
* Field labels, hint text, and associated incidental text are examples of text not covered by this exception.

This is meant to restrict the exception to the controls that subsume the visible text as part of their visually bounded UI.

**Comment by giacomo-petri** [comment]
@aardrian I generally agree with your statement but how should we interpret [example 3](https://codepen.io/Giacomo-Petri/full/zxOqEbv) compared to a [simple disabled button](https://codepen.io/Giacomo-Petri/full/NPrQNxd)?

In example 3 the floating label sits above the input and IMO is perceived as a single UIC.

I'm not opposing the proposal but if we move in that direction I think we should update the definition of UIC accordingly (normative change) to explicitly and consistently exclude "field labels, hint text, and associated incidental text" from the exception, in order to avoid confusion. In this case, example 3 would fail while the simple disabled button would pass.

Note: The [ACT Rules](https://www.w3.org/WAI/standards-guidelines/act/rules/afw4f7/) currently treat the programmatic label as part of the UIC. As a result, this case is considered an exception and, according to the applicability criteria, is classified as inapplicable (see [Inapplicable Example 6](https://www.w3.org/WAI/standards-guidelines/act/rules/afw4f7/#inapplicable-example-6)). Consequently, under the current rules format, tests cannot fail for this scenario for consistent implementation.

**Comment by gundulaniemann** [comment]
@aardrian In fact I do not agree, as Field labels, hint text, and associated incidental text belong to a control and are perceived as one UIC by end user. Thus they fall under the exception.
Even more: If these contrast, it adds to the user's confusion, like there is a label but no perceivable field.
If there is any information to be perceived by the end user, then the disabled state is the wrong state chosen, it should be in read-only or display-only state instead.

**Comment by detlevhfischer** [comment]
@aardrian wrote:

> Field labels, hint text, and associated incidental text are examples of text not covered by this exception.

Maybe add also inputs of type checkbox and radio?

**Comment by giacomo-petri** [comment]
@detlevhfischer, I think this significantly changes the requirement...

To summarize the discussion so far, there appear to be two divergent interpretations:
1. The label is part of the UIC.
2. The label is not part of the UIC (or at least should not be treated as such, since users may need to read it in order to make conscious choices).

The current normative wording is ambiguous due to the phrase "is perceived by users".

Existing proposals
- Proposal 1
    The input field and its programmatically associated label are considered part of the same user interface component
- Proposal 2
    Labels (and similar) are not considered part of the user interface component

Additional proposal (this approach has been usd in the past to address similar situations where consensus could not be reached): adopt proposal 1 (the more permissive interpretation), combined with a best practice recommending a sufficient contrast ratio for associated labels (and similar elements).

**Comment by aardrian** [comment]
@giacomo-petri 

> I generally agree with your statement but how should we interpret [example 3](https://codepen.io/Giacomo-Petri/full/zxOqEbv) compared to a [simple disabled button](https://codepen.io/Giacomo-Petri/full/NPrQNxd)?

Float labels are already a confusing pattern for many users. IME, users confuse those for values all the time. Combining two poor patterns (float labels and disabled fields) is a decision, and I don't feel the float label should drive exception interpretation.

> Note: The [ACT Rules](https://www.w3.org/WAI/standards-guidelines/act/rules/afw4f7/) currently treat the programmatic label as part of the UIC.

Yup, I am aware.

@gundulaniemann 

> In fact I do not agree, as Field labels, hint text, and associated incidental text belong to a control and are perceived as one UIC by end user. 

This treats end users as a single-minded monolith. Would a generic user consider float label (anti-)pattern as a single _thing_? Sure, because it's _designed_ to look like a singular thing, it's most likely to be perceived that way. That was the designer's intent.

A vanilla HTML label and text box? Working with users, I'm not so sure all of them think that is a singular thing since many folks consider the control itself to be distinct from the words.

The [UIC definition](https://www.w3.org/WAI/WCAG22/Understanding/contrast-minimum.html#dfn-user-interface-component) kinda supports this: "…perceived by users as a single control…", "...what the user perceives as separate controls...", "...User interface components include form elements and links..."

Nowhere does it explicitly mention the text associated with the control, let alone a _label_. I would think if a field label was intended to be an exception, the SC would lean on the [normative definition of label](https://www.w3.org/WAI/WCAG22/Understanding/headings-and-labels.html#dfn-label) that we see in 2.4.6. In fact, the [1.4.3 Understanding](https://www.w3.org/WAI/WCAG22/Understanding/contrast-minimum.html#dfn-user-interface-component) doc studiously avoids using the word _label_.

So @giacomo-petri, to your summary, I think "is perceived by users" is less ambiguous when considering the explicit absence of the word "label" or even statements about text.

@detlevhfischer 

> Maybe add also inputs of type checkbox and radio?

Sure. "Field labels" in my statement was meant to cover that but I understand folks might want it to be far more explicit.

BTDubs, I appreciate y'all being willing to jump back into the discussion. I've tried to bring new information / context so it's not a waste of time.

[Edited to properly tag @gundulaniemann so my response to their comment was more clear.]

**Comment by bruce-usab** [comment]
Discussed on TF call 20 Feb. 
- Suggestion made to compare relationship with guidance for Target Size which has a similar difficulty regarding UIC.
- Suggestion made to address similar concepts (inactive, disabled, unavailable, etc.) holistically rather than on a case-by-case basis, one SC at a time.

**Comment by patrickhlauke** [comment]
related https://web-a11y.slack.com/archives/C7KTV5CBA/p1789510327412769
